bin/release.php

```diff
@@ -0,0 +1,283 @@
+	purge_dev_files();
+
+	// Create a new local branch if none exists, else checkout to it
+	$branch_exists = execute_command( sprintf( 'git ls-remote --exit-code --heads origin %s', escapeshellarg( $release_branch_prod ) ), '', true );
+	if ( empty( $branch_exists ) ) {
+		execute_command( sprintf( 'git checkout -b %s', escapeshellarg( $release_branch_prod ) ), "Could not create local release branch: $release_branch_prod" );
+	} else {
+		$remote_url = execute_command( 'git remote get-url --all origin', 'Error', true );
+		execute_command( sprintf( 'git clone --depth 1 -b %1s --single-branch %2s %3s', escapeshellarg( trim( $release_branch_prod ) ), escapeshellarg( $remote_url ), escapeshellarg( $tmp ) ), 'Could not do it!' );
+		execute_command( sprintf( 'rsync -r --delete --exclude="*.git*" . %s', $tmp ), 'Could not rsync.' );
+		chdir( $tmp );
+	}
+
+	// Commit and push!
+	execute_command( 'git commit -am "New Build"', 'Could not commit.' );
+	execute_command( sprintf( 'git push -u origin %s', escapeshellarg( $release_branch_prod ) ), "Could not push $release_branch_prod to remote." );
+}
+
+/**
+ * Resets everything.
+ */
+function reset_and_clean() {
+	execute_command( 'git fetch origin && git checkout master && git reset --hard origin/master' );
+	execute_command( 'rm -rf /tmp/build-release' );
+}
+
+function purge_dev_files() {
+	$ignored = file( dirname( dirname( __FILE__ ) ) . '/.svnignore', FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES );
+
+	foreach ( $ignored as $file_pattern ) {
+		execute_command( sprintf( 'rm -rf %s', $file_pattern ) );
+	}
+}
+
+/**
+ * Prompt for user-input strings
+ *
+ * @param string $question
+ * @param array  $options
+ * @return bool|string
+ */
+function prompt( $question = '', $options = array(), $show_options = true ) {
+	if ( empty( $question ) ) {
+		usage();
+	};
+
+	needs_action( $question );
+	if ( ! empty( $options ) ) {
+		$options_string = 'Your options are: ' . implode( ' or ', $options ) . "\n";
+		if ( $show_options ) {
+			echo $options_string;
+		}
+	}
+
+	$handle = fopen( 'php://stdin', 'r' );
+	$line   = trim( fgets( $handle ) );
+
+	if ( ! empty( $options ) && ! in_array( $line, $options ) ) {
+		fail( 'Sorry, that is not a valid input. Try again?' );
+		echo $options_string;
+		prompt( $question, $options, false );
+	}
+
+	echo "\n";
+	return preg_replace( '/[^A-Za-z0-9\-\.]/', '', $line );
+}
+
+/**
+ * Get a yes/no confirmation
+ *
+ * @param string $question
+ */
+function confirm( $question = '' ) {
+	$question = ! empty( $question ) ? $question : "Are you sure you want to do this?  Type 'yes' to continue: ";
+	needs_action( $question );
+
+	$handle = fopen( 'php://stdin', 'r' );
+	$line   = fgets( $handle );
+	if ( trim( $line ) != 'yes' ) {
+		exit;
+	}
+	echo "\n";
+}
+
+
+// Bold
+function needs_action( $string ) {
+	echo sprintf( '%c[%sm%s%c[0m', 27, 1, (string) $string, 27 );
+}
+
+// Green
+function success( $string ) {
+	printf( "\e[32m%s\e[0m\n", $string );
```

Comment:
Above you casted as string, but not here nor below. Could we cast as string everywhere?